auth_fetch.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-steplib/steps-deploy-to-itunesconnect-deliver/devportalservice"
+)
+
+// AppleAuth contains either Apple ID or APIKey auth info
+type AppleAuth struct {
+	AppleID *AppleIDAuth
+	APIKey  *devportalservice.JWTConnection
+}
+
+// AppleIDAuth contains Apple ID auth info
+type AppleIDAuth struct {
+	// Either session (if using 2FA) or username+password is required
```

Comment:
if Apple ID auth used:
- apple id + password are required
- + appSpecificPassword + TFA session in case of a TFA enabled account

As spaceship uses 
- iTMSTransporter: it requires apple id + password (or appSpecificPassword in case of TFA)
- TunesAPI: it requires apple id + password (+ TFA session in case of TFA)